### PR TITLE
feat: Add fine coregistration script for satellite image fine-tuning

### PR DESCRIPTION
**Pull request by Ritika-K7**
This PR introduces a new script, `georeference.py`, that performs **fine coregistration** on pre-aligned satellite images using the **AROSICS** library.
The script refines spatial alignment by detecting and correcting small misalignments relative to ground truth reference data.

### Changes Made

**New Additions**

* `data_preprocessing/georeference.py` – Script for fine-tuning alignment of satellite images
* `processed_data/results_georeference/` – Output directory for georeferenced images

**Updated Files**

* `config_template.json` – Added entry for georeference output directory
* `load_config.py` – New helper functions to fetch georeference paths from config

### Key Features

* Fine image alignment using **AROSICS**
* Automatically falls back to original image if correction isn't needed
* Batch processes multiple images
* Fully integrated with existing config system

### Technical Details

| Parameter        | Value                                                                           |
| ---------------- | ------------------------------------------------------------------------------- |
| **Input**        | Pre-aligned `.tif` files from `results_batch_align/`                            |
| **Reference**    | `2016_HiRes_Final_Coastline_UTM3N.tif` (Ground truth)                           |
| **Output**       | Georeferenced files with `_GeoRegistered.tif` suffix in `results_georeference/` |
| Method  |AROSICS coregistration with configurable parameters|

### How to Run

```
# 1. Activate your conda environment
conda activate your_environment_name

# 2. Navigate to the project directory
cd CoastlineExtraction/data_preprocessing

# 3. Run the georeferencing script
python georeference.py
```

**Note** : Make sure to replace `your_environment_name` with your actual conda environment.

### Usefulness

* Improves spatial accuracy before model training or coastline extraction
* Automates batch alignment with smart fallback logic
* Helps maintain consistent geographic reference across datasets

### CODE REVIEW

**Comment by fwitmer** [comment]
Looks good. We'll need to remove the satellite image files before merging this branch to the main branch.
